frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js

```diff
@@ -602,6 +602,59 @@ describe("scenarios > dashboard", () => {
       });
     });
   });
+
+  it.skip("should be possible to visit a dashboard with click-behavior linked to the dashboard without permissions (metabase#15368)", () => {
+    cy.server();
+    cy.route("GET", "/api/dashboard/2").as("loadDashboard");
```

Comment:
Is this supposed to be `NEW_DASHBOARD_ID` ?
And I'm not sure if this request should even exist - seems heavy to load each Click Behavior reference.

Reply:
> Is this supposed to be NEW_DASHBOARD_ID ?

It is. I'll have to move `cy.route` within the context of `NEW_DASHBOARD_ID`.

> And I'm not sure if this request should even exist - seems heavy to load each Click Behavior reference.

Not sure either, but it seems this is the only thing I can wait for if I want to avoid false positives.

Reply:
Then let's wait with the test until this is solved. It doesn't seem correct that we make requests for each object - that's an performance issue by itself.

Reply:
@flamber @paulrosenzweig without waiting for anything, it seems test always fails on `37.65`. It originally finds the title "Orders in a dashboard", though.
If you're both ok with it, I can remove any waiting and simplify test a lot?
![image](https://user-images.githubusercontent.com/31325167/112851110-f1c52000-90aa-11eb-85b3-adf1ab0d3ea9.png)

Reply:
Unfortunately, we do need to load the target question/dashboard when viewing a dashboard with click behavior. 

When a card on dashboard 123 has click behavior going to dashboard 456, it saves "456" along with any target filter ids in the visualization settings. IIRC, we pass the parameter value through the url. We need to fetch dashboard 456 to know the filter id => url slug mapping. If dashboard 123 saved the url slug instead of the filter id, it would break if the name is updated.

When the target is a question, we need to fetch the question so we can update the MBQL correctly to add filters when directing users to the QB. Though, we might be able to nest the target question instead of modifying it if we're comfortable with that UX.

I think the additional fetches probably don't add noticeable latency and they should prevent fetches when actually using the click behavior. If the requests slow things down too much, the needed data could be bundled in the data for the originating dashboard.

Reply:
I'm sure it might end up with a race-condition without a wait, since it might have fully loaded `POST /api/card/1/query`
I don't have a great solution for this until the issue is fixed - and not using a request for each object.

Reply:
> without waiting for anything, it seems test always fails on `37.65`

I don't think we can rely on that. The ordering of displaying the data and loading the target dashboard isn't guaranteed.

Reply:
Ok, then we wait for `GET /api/dashboard/2` and see how it'll be solved.